README.md

```diff
@@ -229,6 +280,234 @@ All `VAULT_*` env keys are preserved when using `CONSUL_SERVICE_TAG`, `address`
+  "db_max_ttl": "72h",
+  "environment_name": "staging",
+  "environment_tld": "stag",
+  "here_doc": "something multiline\nthat will be available\nas a single string",
+  "stuff": [
+    "a",
+    "b",
+    "c"
+  ]
+}
+```
+
+### Functions
+
+This tool implement most of the features available in [hashicorp/consul-template](https://github.com/hashicorp/consul-template).
+
+Additionally some useful functions in the context of hashi-helper is available, please see the list further down
+
+#### consul-template compatability
+
+The functions currently implemented can be found below.
+
+- [base64Decode](https://github.com/hashicorp/consul-template#base64Decode)
+- [base64Encode](https://github.com/hashicorp/consul-template#base64Encode)
+- [base64URLDecode](https://github.com/hashicorp/consul-template#base64URLDecode)
+- [base64URLEncode](https://github.com/hashicorp/consul-template#base64URLEncode)
+- [contains](https://github.com/hashicorp/consul-template#contains)
+- [containsAll](https://github.com/hashicorp/consul-template#containsAll)
+- [containsAny](https://github.com/hashicorp/consul-template#containsAny)
+- [containsNone](https://github.com/hashicorp/consul-template#containsNone)
+- [containsNotAll](https://github.com/hashicorp/consul-template#containsNotAll)
+- [env](https://github.com/hashicorp/consul-template#env)
+- [in](https://github.com/hashicorp/consul-template#in)
+- [join](https://github.com/hashicorp/consul-template#join)
+- [parseBool](https://github.com/hashicorp/consul-template#parseBool)
+- [parseFloat](https://github.com/hashicorp/consul-template#parseFloat)
+- [parseInt](https://github.com/hashicorp/consul-template#parseInt)
+- [parseJSON](https://github.com/hashicorp/consul-template#parseJSON)
+- [parseUint](https://github.com/hashicorp/consul-template#parseUint)
+- [plugin](https://github.com/hashicorp/consul-template#plugin)
+- [regexMatch](https://github.com/hashicorp/consul-template#regexMatch)
+- [regexReplaceAll](https://github.com/hashicorp/consul-template#regexReplaceAll)
+- [replaceAll](https://github.com/hashicorp/consul-template#replaceAll)
+- [scratch](https://github.com/hashicorp/consul-template#scratch)
+- [split](https://github.com/hashicorp/consul-template#split)
+- [timestamp](https://github.com/hashicorp/consul-template#timestamp)
+- [toJSON](https://github.com/hashicorp/consul-template#toJSON)
+- [toJSONPretty](https://github.com/hashicorp/consul-template#toJSONPretty)
+- [toLower](https://github.com/hashicorp/consul-template#toLower)
+- [toTitle](https://github.com/hashicorp/consul-template#toTitle)
+- [toUpper](https://github.com/hashicorp/consul-template#toUpper)
+- [toYAML](https://github.com/hashicorp/consul-template#toYAML)
+- [trimSpace](https://github.com/hashicorp/consul-template#trimSpace)
+
+#### lookup
+
+`lookup` is used to lookup template variables inside a template. If the key do not exist, the template rendering will fail.
+
+`[[ lookup "my-key" ]]` will output `hello-world`
```

Comment:
It would be useful to add `my-key` to the examples of variable files, so that the example is runnable.